java-api-core/src/com/kinvey/java/sync/dto/SyncRequest.java

```diff
@@ -49,9 +49,11 @@
         }
 
         public static HttpVerb fromString(String verb){
-            for (HttpVerb v : HttpVerb.values()){
-                if (v.query.equals(verb)){
-                    return v;
+            if (verb != null) {
+                for (HttpVerb v : HttpVerb.values()){
+                    if (v.query.equals(verb)){
```

Comment:
I'm wondering if this check needs to be case insensitive

Reply:
Added fix to make this compare logic case insensitive.